src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Extensions/HttpRequestExtensionsTests.cs

```diff
@@ -0,0 +1,261 @@
+        }
+
+        private static async Task AssertHttpRequestsWithQueryParameters(string queryString, IReadOnlyList<Tuple<string, string>> expectedParameters)
+        {
+            // Validate parameters for not encoded URI.
+            HttpRequest rawRequest = await GetHttpRequestAsync(queryString);
+            var rawRequestParameters = rawRequest.GetQueriesForSearch();
+            Assert.Equal(expectedParameters.Count, rawRequestParameters.Count);
+            AssertQueryParameters(expectedParameters, rawRequestParameters);
+
+            // Check compatibility with previous query string parameter identification approach.
+            var previousRequestParameters = GetPreviousQueriesForSearchApproach(rawRequest);
+            Assert.Equal(expectedParameters.Count, previousRequestParameters.Count);
+            AssertQueryParameters(expectedParameters, previousRequestParameters);
+
+            // Validate parameters for encoded URI.
+            string encodedQueryString = HttpFhirUtility.EncodeUrl(queryString);
+            HttpRequest encodedRequest = await GetHttpRequestAsync(encodedQueryString);
+            var encodedRequestParameters = encodedRequest.GetQueriesForSearch();
+            Assert.Equal(expectedParameters.Count, encodedRequestParameters.Count);
+            AssertQueryParameters(expectedParameters, encodedRequestParameters);
+        }
+
+        private static void AssertQueryParameters(IReadOnlyList<Tuple<string, string>> expected, IReadOnlyList<Tuple<string, string>> actual)
+        {
+            Assert.NotNull(expected);
+
+            Assert.NotNull(actual);
+
+            Assert.Equal(expected.Count, actual.Count);
+
+            for (int i = 0; i < expected.Count; i++)
+            {
+                Assert.Equal(expected[i].Item1, actual[i].Item1);
+                Assert.Equal(expected[i].Item2, actual[i].Item2);
+            }
+        }
+
+        private static async Task<HttpRequest> GetHttpRequestAsync(string queryString)
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Host = new HostString("localhost");
+            httpContext.Request.ContentType = "application/json";
+            if (!string.IsNullOrWhiteSpace(queryString))
+            {
+                httpContext.Request.QueryString = new QueryString(string.Concat("?", queryString));
+            }
+
+            using (var stream = new MemoryStream())
+            using (var writer = new StreamWriter(stream))
+            {
+                await writer.WriteAsync("{}");
+                await writer.FlushAsync();
+                stream.Position = 0;
+                httpContext.Request.Body = stream;
+
+                return httpContext.Request;
+            }
+        }
+
+        /// <summary>
+        /// This methods runs the previous approach adopted to extract query string parameters from an URI.
```

Comment:
nit methods -> method

Reply:
Just for my understanding, are we planning to decommission this test compare previsous vs new approach once the new approach gets stable?